src/tyr/route_serializer_osrm.cc

```diff
@@ -283,23 +206,85 @@ json::MapPtr geojson_shape(const std::vector<PointLL> shape) {
   return geojson;
 }
 
+// Maps Valhalla edge and road class to an OpenLR form-of-way.
+FormOfWay RoadClassToFOW(const TripLeg::Node& node) {
+  const bool is_missing_rc = !node.has_edge() || !node.edge().has_road_class();
+  if (is_missing_rc || !node.edge().has_use()) {
+    return FormOfWay::OTHER;
+  }
+  const TripLeg::Use use = node.edge().use();
+  const valhalla::RoadClass rc = node.edge().road_class();
+  const bool one_way =
+      node.edge().has_traversability() && node.edge().traversability() != TripLeg::kBoth;
+  if (node.edge().roundabout()) {
+    return FormOfWay::ROUNDABOUT;
+  } else if (use == TripLeg::kRampUse || use == TripLeg::kTurnChannelUse) {
+    return FormOfWay::SLIPROAD;
+  } else if (rc == valhalla::RoadClass::kMotorway && one_way) {
+    return FormOfWay::MOTORWAY;
+  } else if (rc <= valhalla::RoadClass::kTertiary && one_way) {
+    return FormOfWay::MULTIPLE_CARRIAGEWAY;
+  } else if (rc <= valhalla::RoadClass::kTertiary) {
+    return FormOfWay::SINGLE_CARRIAGEWAY;
+  } else {
+    return FormOfWay::OTHER;
+  }
+}
+
+// TODO(mookerji): Move this shared header file somewhere and remove also from tests/openlr.cc
```

Comment:
yeah i feel like we may as well just move this to the openlr header itself and provide an option to the `toBinary(bool b64_encode = false)` method to optionally b64 encode the results

Reply:
same with the constructor that takes the binary reference. should become `LinearReference(const std::string& reference, bool b64_encoded = false)`

Reply:
Done